presto-native-execution/CMakeLists.txt

```diff
@@ -103,81 +117,88 @@ set(VELOX_BUILD_TEST_UTILS
-  program_options
-  context
-  filesystem
-  regex
-  thread
-  system
-  date_time
-  atomic)
-include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
-
-find_package(gflags COMPONENTS shared)
+set(BOOST_INCLUDE_LIBRARIES
+    headers
+    atomic
+    context
+    date_time
+    filesystem
+    program_options
+    regex
+    system
+    thread)
+
+set_source(Boost)
+resolve_dependency(Boost 1.66.0 COMPONENTS ${BOOST_INCLUDE_LIBRARIES})
+
+set_source(gflags)
+resolve_dependency(gflags COMPONENTS shared)
 
+# TODO
 find_library(ANTLR4_RUNTIME antlr4-runtime)
 find_path(
   ANTLR4_RUNTIME_INCLUDE_DIR
   NAMES antlr4-runtime.h
   PATH_SUFFIXES antlr4-runtime)
 include_directories(SYSTEM ${ANTLR4_RUNTIME_INCLUDE_DIR})
 
-find_library(GLOG glog)
+if(${gflags_SOURCE} STREQUAL "BUNDLED")
+  # we force glog from source to avoid issues with a system version built
+  # against another gflags version (which is likely)
+  set(glog_SOURCE BUNDLED)
+else()
+  set(glog_SOURCE SYSTEM)
+endif()
+resolve_dependency(glog)
 
-find_library(FMT fmt)
+set_source(fmt)
+resolve_dependency(fmt)
 
+# TODO
 find_library(EVENT event)
 
-find_library(DOUBLE_CONVERSION double-conversion)
-
-find_library(LZ4 lz4)
-find_library(LZO lzo2)
-find_library(ZSTD zstd)
-find_package(ZLIB)
-find_library(SNAPPY snappy)
-
-find_package(folly CONFIG REQUIRED)
-set(FOLLY_WITH_DEPENDENCIES
-    ${FOLLY_LIBRARIES}
-    ${DOUBLE_CONVERSION}
-    Boost::context
-    dl
-    ${EVENT}
-    ${SNAPPY}
-    ${LZ4}
-    ${ZSTD}
-    ${ZLIB_LIBRARIES})
-
-find_package(BZip2 MODULE)
-if(BZIP2_FOUND)
-  list(APPEND FOLLY_WITH_DEPENDENCIES ${BZIP2_LIBRARIES})
-endif()
-include_directories(SYSTEM ${FOLLY_INCLUDE_DIRS})
+set_source(double-conversion)
+resolve_dependency(double-conversion)
+
+set_source(lz4)
+resolve_dependency(lz4)
+set_source(lzo2)
+resolve_dependency(lzo2)
+set_source(zstd)
+resolve_dependency(zstd)
+set_source(ZLIB)
+resolve_dependency(ZLIB)
+set_source(Snappy)
+resolve_dependency(Snappy)
+
+# Locate or build folly.
+add_compile_definitions(FOLLY_HAVE_INT128_T=1)
+set_source(folly)
+resolve_dependency(folly)
 
 # Include third party header files
 find_path(OPT_OPENSSL_DIR NAMES opt/openssl@1.1)
 set(OPENSSL_ROOT_DIR "${OPT_OPENSSL_DIR}/opt/openssl@1.1")
-find_package(OpenSSL REQUIRED)
-
-find_library(LIBSODIUM_LIBRARY NAMES sodium)
-find_library(PROXYGEN proxygen)
-find_library(PROXYGEN_HTTP_SERVER proxygenhttpserver)
-find_library(FIZZ fizz)
-find_library(WANGLE wangle)
-
+find_package(OpenSSL REQUIRED)  # TODO
+
+find_library(LIBSODIUM_LIBRARY NAMES sodium)  # TODO
+set_source(proxygen)
+resolve_dependency(proxygen)
+find_library(PROXYGEN_HTTP_SERVER proxygenhttpserver)  # TODO
+set_source(fizz)
+resolve_dependency(fizz)
+set_source(wangle)
+resolve_dependency(wangle)
+
+# TODO: the following works, but you need to patch Findre2.cmake in velox,
+# to use NOT TARGET in the last if block and move two `set` calls above it
+# set_source(re2)
+# resolve_dependency(re2)
 find_library(RE2 re2)
```

Comment:
Will submit a fix to velox once this PR is approved.

Reply:
Should we add new files like `double-conversion.cmake` etc. to Presto only? They won't be used by velox.